libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts

```diff
@@ -419,17 +433,18 @@ export class AdaptiveDialog<O extends object = {}> extends DialogContainer<O> im
 
     /**
      * Creates a child [DialogContext](xref:botbuilder-dialogs.DialogContext) for the given context.
+     *
      * @param dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
      * @returns The child [DialogContext](xref:botbuilder-dialogs.DialogContext) or null if no [AdaptiveDialogState.actions](xref:botbuilder-dialogs-adaptive.AdaptiveDialogState.actions) are found for the given context.
      */
     public createChildContext(dc: DialogContext): DialogContext {
         const activeDialogState = dc.activeDialog.state;
-        let state: AdaptiveDialogState = activeDialogState[this.adaptiveKey];
+        const state: AdaptiveDialogState = activeDialogState[this.adaptiveKey];
         if (!state) {
-            state = { actions: [] };
+            activeDialogState[this.adaptiveKey] = { actions: [] };
```

Comment:
I don't think this actually propagates back to `state` (see [Typescript playground sample](https://www.typescriptlang.org/play?#code/MYewdgzgLgBA5gJxAVwA4C4YCUCmoEAmAPNAgJZhwA0MyYA1mCAO5gB8MAvDAN4C+AbgBQQ0JFj0cATy4wA5JKlzho8NBjQAhlByzEKVAG1FAXWFkAZjAAUAQi06AlLyEw38JGmPSTsnosw5ADdNABtkHDlBIT4RMQgQUJwAOlCQOGsHHEcBIA)).

```typescript
activeDialogStatel[this.adaptiveKey] ??= {actions: []};
const state = activeDialogState[this.adaptiveKey];
```

Reply:
Correct. In this method, it doesn't need to since `state` isn't used in other blocks. I'll change the next `if` to an `else if` to make that more clear.